src/libServer/IsolatedServer.cpp

```diff
@@ -491,6 +549,166 @@ Json::Value IsolatedServer::CreateTransaction(const Json::Value& _json) {
   }
 }
 
+Json::Value IsolatedServer::CreateTransactionEth(EthFields const& fields,
```

Comment:
Presumably challenging, but do you think it's feasible to split this function into smaller entities (now it's > 150 loc).

Reply:
yes, I can do this - I was basically copying the code from the scilla way of adding TX so can refactor both in a seperate PR if you like?